transport/x/grpc/outbound.go

```diff
@@ -186,6 +187,14 @@ func (o *Outbound) invoke(
 	)
 }
 
+func metadataToIsApplicationError(responseMD metadata.MD) bool {
+	if responseMD == nil {
+		return false
+	}
+	value, ok := responseMD[ApplicationErrorHeader]
```

Comment:
This makes an implicit assumption that the absence of the ApplicationErrorHeader implies that there is no application error, where it's possible we're dealing with an "Unknown" case, we really don't know whether there is an application error.

Still an x package, and we don't really have much action if it's not there anyway, so I'm ok leaving it for now, just wanted to bring it up.

Reply:
I see your point, but YARPC as a whole doesn't have a concept of dealing with Unknown values for headers, and I don't know what action would be taken if the value was Unknown (any proposals?). I actually think this is the best definition, as opposed to an actual success and error value - the presence of the header indicates error, there's no ambiguity.

Reply:
Just to document for later - I think this is what it should be when we (soon) pull this out of x/. Additionally, this is implicitly what the HTTP transport does - if there is an "error" value on the HTTP transport, it's an error, even though there is a "success" header defined, it's never checked against, which means the HTTP error documents two values but only uses one for decision making.